core/src/main/java/org/mskcc/cbio/portal/dao/JdbcDataSource.java

```diff
@@ -3,28 +3,45 @@
 import org.apache.commons.dbcp2.BasicDataSource;
 import org.mskcc.cbio.portal.util.DatabaseProperties;
 import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.Assert;
 
 /**
  * Data source that self-initializes based on cBioPortal configuration.
  */
 public class JdbcDataSource extends BasicDataSource {
     public JdbcDataSource () {
         DatabaseProperties dbProperties = DatabaseProperties.getInstance();
+
         String host = dbProperties.getDbHost();
         String userName = dbProperties.getDbUser();
         String password = dbProperties.getDbPassword();
         String mysqlDriverClassName = dbProperties.getDbDriverClassName();
         String database = dbProperties.getDbName();
         String useSSL = (!StringUtils.isBlank(dbProperties.getDbUseSSL())) ? dbProperties.getDbUseSSL() : "false";
         String enablePooling = (!StringUtils.isBlank(dbProperties.getDbEnablePooling())) ? dbProperties.getDbEnablePooling(): "false";
-        String url ="jdbc:mysql://" + host + "/" + database +
-                        "?user=" + userName + "&password=" + password +
-                        "&zeroDateTimeBehavior=convertToNull&useSSL=" + useSSL;
+        String connectionURL = dbProperties.getConnectionURL();
+        
+        Assert.hasText(userName, errorMessage("username", "db.user"));
+        Assert.hasText(password, errorMessage("password", "db.password"));
+        
+        // For backward compatibility, build connection URL from individual properties.
+        if (connectionURL == null) {
+            Assert.hasText(host, errorMessage("host", "db.host") + 
+                " Or preferably, set the 'db.connection_string' and remove 'db.host', 'db.database' and 'db.use_ssl' (best practice).");
+            Assert.hasText(database, errorMessage("database", "db.database") + 
+                " Or preferably, set the 'db.connection_string' and remove 'db.host', 'db.database' and 'db.use_ssl (best practice).");
+            connectionURL = String.format(
+                "jdbc:mysql://%s/%s&zeroDateTimeBehavior=convertToNull&useSSL=%s",
```

Comment:
@pvannierop This may be because I updated my local JDBC driver to 8.0.31, but I also had to add **allowLoadLocalInfile=true** to allow the bulk loader to work properly.

Reply:
@n1zea144 The idea of this PR is that when you have the need for these params, you should provide the full connection string with the `d.connection_string` parameter. Remember that `allowLoadLocalInfile=true` is considered as a security risk. Activation as a default is not the way to go in my opinion.

Reply:
re `allowLoadLocalInfile=true` - I guess we would apply this just for the importer? Doesn't seem like much of a security risk in that setting

Reply:
>I guess we would apply this just for the importer
Correct